src/service/extension-location.js

```diff
@@ -59,15 +68,21 @@ export function calculateExtensionScriptUrl(
   opt_extensionVersion,
   opt_isLocalDev
 ) {
-  const base = calculateScriptBaseUrl(location, opt_isLocalDev);
-  const rtv = getMode().rtvVersion;
+  const useDefaultCdn =
+    isCacheModifiedExtension(extensionId) &&
+    !urls.cdnSupportsCacheModifiedExtensions;
+  const base = calculateScriptBaseUrl(location, opt_isLocalDev, useDefaultCdn);
   if (opt_extensionVersion == null) {
     opt_extensionVersion = '0.1';
   }
   const extensionVersion = opt_extensionVersion
     ? '-' + opt_extensionVersion
     : '';
-  return `${base}/rtv/${rtv}/v0/${extensionId}${extensionVersion}.js`;
+  let extensionPath = `v0/${extensionId}${extensionVersion}.js`;
+  if (urls.cdn === urls.defaultCdn || useDefaultCdn) {
+    extensionPath = `rtv/${getMode().rtvVersion}/${extensionPath}`;
+  }
+  return `${base}/${extensionPath}`;
```

Comment:
For https://github.com/ampproject/amphtml/issues/25214#issuecomment-545630142, I was actually hoping to return just `${base}/${extensionPath}` if `opt_isLocalDev` or `getMode().test` is true. Is that a change we can make?

Reply:
/CC @honeybadgerdontcare @sebastianbenz Would you two mind taking @cathyxz's request into consideration in your design doc?